### PR TITLE
Correct warning logging when update lock is present for git_pillar/winrepo, add runner function for clearing git_pillar/winrepo locks

### LINKED ISSUE #27738: Git Pillar locks not managed by fileserver runner
I receive warnings in the master log file similar to the below:

`Update lockfile is present for git_pillar remote 'master git@bitbucket.org:repo.git', skipping. If this warning persists, it is possible that the update process was interrupted. Removing /var/cache/salt/master/git_pillar/b2f9cc5623c6bfbe65a310fb7ef60a90/update.lk or running 'salt-run fileserver.clear_lock git_pillar' will allow updates to continue for this remote.`

However, when I run fileserver.clear_lock, it doesn't unlock git_pillar, only gitfs. This is the same (as far as I can tell) for the entire fileserver module. 

salt 2015.8.0-1809-g611d3e5 (Beryllium) on CentOS 7.

### PR DESCRIPTION
**Pull request by terminalmage**
Fixes #27738.

### CODE REVIEW

File: salt/fileserver/__init__.py [hunk 1]
```diff
@@ -272,6 +272,34 @@ def is_file_ignored(opts, fname):
     return False
 
 
+def clear_lock(clear_func, lock_type, remote=None):
+    '''
+    Function to allow non-fileserver functions to clear update locks
```
Comment (rallytime) [inline salt/fileserver/__init__.py]: Versionadded here too?

File: salt/fileserver/__init__.py [hunk 2]
```diff
@@ -272,6 +272,34 @@ def is_file_ignored(opts, fname):
...
+        that a subset of remotes can be targeted.
+
+
+    Returns a tuple containing two lists: One with messages
+    '''
+    msg = 'Clearing update lock for {0} remotes'.format(lock_type)
+    if remote:
+        msg += ' matching {0}'.format(remote)
+    log.debug(msg)
+    return clear_func(remote=remote)
```
Comment (rallytime) [inline salt/fileserver/__init__.py]: Should this be `clear_func` or `clear_lock`?

**Comment by rallytime** [comment]
Looks like there is also some small [lint errors](https://jenkins.saltstack.com/job/salt-pr-lint-n/9861/violations/file/salt/runners/cache.py/).

File: salt/fileserver/__init__.py (hunk 1 above)
Comment (terminalmage) [inline salt/fileserver/__init__.py]: This function is not exposed to the docs and can't be executed directly from the CLI, so I don't see a benefit for adding one.

**Comment by terminalmage** [comment]
@rallytime There was actually a legit issue found in the lint failure, I wasn't initializing the WinRepo instance properly, and I also noticed that I wasn't accounting for the new "ng" winrepo config paths. These are now fixed, I confirmed by manually adding an update.lk and then using the runner function to clear it.

**Pull request merged**